1.segment_images/scripts/1.segment_cells_organoids.py

```diff
@@ -0,0 +1,291 @@
+
+
+original_nuclei_image = nuclei.copy()
+original_cyto_image = cyto.copy()
+
+original_nuclei_z_count = nuclei.shape[0]
+original_cyto_z_count = cyto.shape[0]
+
+
+# In[5]:
+
+
+# make a 2.5 D max projection image stack with a sliding window of 3 slices
+image_stack_2_5D = np.empty((0, cyto.shape[1], cyto.shape[2]), dtype=cyto.dtype)
+for image_index in range(cyto.shape[0]):
+    image_stack_window = cyto[image_index : image_index + window_size]
+    if not image_stack_window.shape[0] == window_size:
+        break
+    # max project the image stack
+    image_stack_2_5D = np.append(
+        image_stack_2_5D, np.max(image_stack_window, axis=0)[np.newaxis, :, :], axis=0
+    )
+
+image_stack_2_5D = np.array(image_stack_2_5D)
+cyto = np.array(image_stack_2_5D)
+print("2.5D cyto image stack shape:", cyto.shape)
+
+
+# make a 2.5 D max projection image stack with a sliding window of 3 slices
+image_stack_2_5D = np.empty((0, nuclei.shape[1], nuclei.shape[2]), dtype=nuclei.dtype)
+for image_index in range(nuclei.shape[0]):
+    image_stack_window = nuclei[image_index : image_index + window_size]
+    if not image_stack_window.shape[0] == window_size:
+        break
+    # max project the image stack
+    image_stack_2_5D = np.append(
+        image_stack_2_5D, np.max(image_stack_window, axis=0)[np.newaxis, :, :], axis=0
+    )
+
+image_stack_2_5D = np.array(image_stack_2_5D)
+nuclei = np.array(image_stack_2_5D)
+print("2.5D nuclei image stack shape:", nuclei.shape)
+
+
+# In[6]:
+
+
+if in_notebook:
+    # plot the nuclei and the cyto channels
+    plt.figure(figsize=(10, 10))
+    plt.subplot(121)
+    plt.imshow(nuclei[9, :, :], cmap="gray")
+    plt.title("nuclei")
+    plt.axis("off")
+    plt.subplot(122)
+    plt.imshow(cyto[9, :, :], cmap="gray")
+    plt.title("cyto")
+    plt.axis("off")
+    plt.show()
+
+
+# In[7]:
+
+
+imgs = []
+# save each z-slice as an RGB png
+for z in range(nuclei.shape[0]):
+
+    nuclei_tmp = nuclei[z, :, :]
+    cyto_tmp = cyto[z, :, :]
+    nuclei_tmp = (nuclei_tmp / nuclei_tmp.max() * 255).astype(np.uint8)
+    cyto_tmp = (cyto_tmp / cyto_tmp.max() * 255).astype(np.uint8)
+    # save the image as an RGB png with nuclei in blue and cytoplasm in red
+    RGB = np.stack([cyto_tmp, np.zeros_like(cyto_tmp), nuclei_tmp], axis=-1)
+
+    # change to 8-bit
+    RGB = (RGB / RGB.max() * 255).astype(np.uint8)
+
+    rgb_image_pil = Image.fromarray(RGB)
+
+    imgs.append(rgb_image_pil)
+
+
+# ## Cellpose
+
+# In[8]:
+
+
+# model_type='cyto' or 'nuclei' or 'cyto2' or 'cyto3'
+model = models.Cellpose(model_type="cyto3", gpu=use_GPU)
+
+channels = [[1, 3]]  # channels=[red cells, blue nuclei]
+diameter = 200
+
+masks_all_dict = {"masks": [], "imgs": []}
+imgs = np.array(imgs)
+
+for img in imgs:
+    masks, flows, styles, diams = model.eval(img, diameter=diameter, channels=channels)
+    masks_all_dict["masks"].append(masks)
+    masks_all_dict["imgs"].append(img)
+print(len(masks_all_dict))
```

Comment:
Consider adding a comment here of what you are doing with the masks and images

Reply:
added